src/test/groovy/graphql/normalized/ExecutableNormalizedOperationToAstCompilerDeferTest.groovy

```diff
@@ -0,0 +1,448 @@
+                }
+                ... on Dog @defer {
+                    breed
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, fields, noVariables)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  animal {
+    ... on Cat @defer {
+      breed
+    }
+    ... on Dog @defer {
+      breed
+    }
+  }
+}
+'''
+    }
+
+    def "nested defer"() {
+        String query = """
+          query q {
+            animal {
+                ... on Cat @defer {
+                    name
+                }
+                ... on Animal @defer {
+                    name
+                    ... on Dog @defer {
+                        owner {
+                            firstname
+                            ... @defer {
+                                lastname
+                            }
+                            ... @defer {
+                                bestFriend {
+                                    firstname
+                                    ... @defer {
+                                        lastname
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, fields, noVariables)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
+  animal {
+    ... @defer {
+      name
+    }
+    ... on Dog @defer {
+      owner {
+        firstname
+        ... @defer {
+          bestFriend {
+            firstname
+            ... @defer {
+              lastname
+            }
+          }
+          lastname
+        }
+      }
+    }
+  }
+}
+'''
+    }
+
+    def "multiple defers at the same level are preserved"() {
+        String query = """
+          query q {
+            dog {
+                ... @defer {
+                    name
+                }
+                ... @defer {
+                    breed
+                }
+                ... @defer {
+                    owner {
+                        firstname
+                    }
+                }
+            }
+          }
+        """
+        GraphQLSchema schema = mkSchema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def result = compileToDocumentWithDeferSupport(schema, QUERY, null, fields, noVariables)
+        def printed = AstPrinter.printAst(new AstSorter().sort(result.document))
+        then:
+        printed == '''{
```

Comment:
this test is currently failing. This is the result:
```
{
  dog {
    ... @defer {
      breed
      name
      owner {
        firstname
      }
    }
  }
}
```

Reply:
I see 2 ways going forward:
- introduce a new class to represent a @defer block. This new class would share a type with ExecutableNormalizedField , so the children of an ENF could either be other ENFs, or @defer blocks.
- Build a map that associates fields with @defer outside of the ENF hierarchy, similarly to normalizedFieldToQueryDirectives.
I'm leaning towards option #2 because it seems less intrusive